speechly/sal/v1/compiler.proto

```diff
@@ -92,3 +103,8 @@ message LineReference {
   // The contents of the message.
   string message = 5;
 }
+
+message ExtractSALResult {
+  // SAL templates
+  string templates = 1;
```

Comment:
Shouldn't this be a repeated field? I don't think it makes sense to return all templates as a single string. Alternatively, we could have a stream of results if you want to implement a streaming API.

Reply:
I agree that streaming would make sense if it is used for very large files. But the specific use case I have, the files shouldn't be very large. Also in my use case, the SAL templates are not parsed nor compiled, like when calling Compile RPC and returning CompileResult, but only the string representation of SAL is extracted from the input. Therefore it makes sense to me to return it as a single string.

Reply:
Okay, it seems I misunderstand the semantics of this API. Do you mind providing some examples of how this new RPC will work?

Reply:
Sure. It takes a tar package, similar to what cli provides at deploy command. Then it extracts the tar, parses the yaml file inside, extracts the SAL templates from the yaml configuration file and returns the SAL templates. The SAL template could be for example "action = [turn on | turn off] \n $action".  I wouldn't like to ExtractSALResult's `templates` to be split line by line, like "action = [turn on | turn off]", "$action" because then I would need to combine those lines again together later. 
Of course I can combine them and if we think that it is more general approach, I'm ok with that too.

Reply:
Aha, I see, thanks!

If this is the case I would like you to change the RPC to bidirectional streaming, since that is the guideline when working with any SAL source uploads / downloads (see for example the Config API). The reasoning behind this is such that if a service is to implement this RPC as a client-side streaming RPC, there is a possibility of a client sending large enough source file that wouldn't fit into the memory on the server-side.

Also, how about renaming the method to something like `ExtractSources` or `ExtractSALSources`? At least to me this makes it a bit more descriptive.

Lastly, I really appreciate you explaining the semantics of the RPC - how about adding them to proto file as a comment on the RPC itself?